command/kv/get/kv_get_test.go

```diff
@@ -418,3 +419,115 @@ func TestKVGetCommand_DetailedBase64(t *testing.T) {
 		t.Fatalf("bad %#v, value is not base64 encoded", output)
 	}
 }
+
+func TestKVGetCommand_KeysRecurse(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	t.Parallel()
+	a := agent.NewTestAgent(t, ``)
+	defer a.Shutdown()
+	client := a.Client()
+
+	ui := cli.NewMockUi()
+	c := New(ui)
+	keys := map[string]string{
+		"foo/":   "",
+		"foo/a":  "Hello World 2",
+		"foo1/a": "Hello World 1",
+	}
+	for k, v := range keys {
+		var pair *api.KVPair
+		switch k {
+		case "":
+			pair = &api.KVPair{Key: k, Value: nil}
+		default:
+			pair = &api.KVPair{Key: k, Value: []byte(v)}
+		}
+		if _, err := client.KV().Put(pair, nil); err != nil {
+			t.Fatalf("err: %#v", err)
+		}
+	}
+	args := []string{
+		"-http-addr=" + a.HTTPAddr(),
+		"-recurse",
+		"-keys",
+		"foo",
+	}
+
+	code := c.Run(args)
+	if code != 0 {
+		t.Fatalf("bad: %d. %#v", code, ui.ErrorWriter.String())
+	}
+	output := ui.OutputWriter.String()
+	fmt.Println(output)
+	for key, value := range keys {
+		if !strings.Contains(output, key) {
+			t.Fatalf("bad %#v missing %q", output, key)
+		}
+		if strings.Contains(output, key+":"+base64.StdEncoding.EncodeToString([]byte(value))) {
```

Comment:
Do you need to base64 the value here when you're not using `-base64` in the `args`?